pkg/phases/kernel.go

```diff
@@ -0,0 +1,127 @@
+	var os OS
+	var err error
+        if sys.Kernel == "" {
+		return []types.Command{}, files, nil
+	}
+	if os, err = GetOSForTag(ctx.Flags...); err != nil {
+		return nil, nil, err
+        }
+        kernelpackage, kernelheaderpackage := os.GetKernelPackageNames(sys.Kernel)
+        commands.Append(os.GetPackageManager().Install(kernelpackage, kernelheaderpackage))
+        // Possible Enhancement, implement this logic in an os.GetGrubManager()
+        return commands.Merge(), files, nil
+}
+
+func (p kernel) Verify(cfg *types.Config, results *types.VerifyResults, flags ...types.Flag) bool {
+	verify := true
+	var os OS
+        var err error
+	if os, err = GetOSForTag(flags...); err != nil {
+		results.Fail("Unable to find OS for tags %s", flags)
+		return false
+	}
+        if cfg.Kernel == "" {
+		return verify
+	}
+        kernelpackage, kernelheaderpackage := os.GetKernelPackageNames(cfg.Kernel)
+        installed := os.GetPackageManager().GetInstalledVersion(kernelpackage)
+        if strings.Contains(installed, cfg.Kernel) {
+		results.Pass("%s is installed", kernelpackage)
+        } else {
+		results.Fail("%s is not installed", kernelpackage)
+                verify = verify && false
+        }
+        installed = os.GetPackageManager().GetInstalledVersion(kernelheaderpackage)
+        if strings.Contains(installed, cfg.Kernel) {
+		results.Pass("%s is installed", kernelheaderpackage)
+        } else {
+		results.Fail("%s is not installed", kernelheaderpackage)
+                verify = verify && false
+        }
+        if os.VerifyDefaultKernel(cfg.Kernel) {
+		results.Pass("%s is the default kernel", cfg.Kernel)
+        } else {
+		results.Fail("%s is not the default kernel", cfg.Kernel)
+                verify = verify && false
+        }
+
+	return verify
+}
+func (p kernel) ProcessFlags(sys *types.Config, flags ...types.Flag) {
+	//All the stuff is done in commands minification?
+}
+
+//This interface and the two corresponding structs are used to provide OS interface easy interface to the common naming conventions
+type KernelNameManager interface {
+        GetKernelPackageNames(version string) (string, string)
+}
+
+// Generally used by yum based systems
+type KernelTypeNaming struct{}
+
+func (p KernelTypeNaming) GetKernelPackageNames(version string) (string, string) {
+        kernelname := fmt.Sprintf("kernel-%s", version)
+        headername := fmt.Sprintf("kernel-devel-%s", version)
+        return kernelname, headername
+}
+
+// Generally used by apt based systems
+type LinuxTypeNaming struct{}
+
+func (p LinuxTypeNaming) GetKernelPackageNames(version string) (string, string) {
+        kernelname := fmt.Sprintf("linux-image-%s", version)
+        headername := fmt.Sprintf("linux-headers-%s", version)
+        return kernelname, headername
+}
+
+// Interface and corresponding structs for reading/writing grub config with grub.conf or grubby
+type GrubConfigManager interface {
+        GetDefaultKernel() (string, bool)
```

Comment:
```suggestion
        ReadDefaultKernel() (string, bool)
```
Getters and Setters usually shouldn't have side-effects e.g. reading from the filesystem etc..